modules/gov/utils_events.go

```diff
@@ -0,0 +1,63 @@
+package gov
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	eventsutil "github.com/forbole/bdjuno/v4/utils/events"
+)
+
+// ProposalIDFromEvent returns the proposal id from the given events
+func ProposalIDFromEvents(events sdk.StringEvents) (uint64, error) {
+	for _, event := range events {
+		attribute, ok := eventsutil.FindAttributeByKey(event, govtypes.AttributeKeyProposalID)
+		if ok {
+			return strconv.ParseUint(attribute.Value, 10, 64)
+		}
+	}
+
+	return 0, fmt.Errorf("no proposal id found")
+}
+
+// VoteOptionFromEvents returns the vote option from the given events
+func VoteOptionFromEvents(events sdk.StringEvents) (govtypesv1.VoteOption, error) {
+	for _, event := range events {
+		attribute, ok := eventsutil.FindAttributeByKey(event, govtypes.AttributeKeyOption)
+		if ok {
+			return parseVoteOption(attribute.Value)
+		}
+	}
+
+	return 0, fmt.Errorf("no vote option found")
+}
+
+// parseVoteOption returns the vote option from the given string
+// option value in string could be 2 cases, for example:
+// 1. "{\"option\":1,\"weight\":\"1.000000000000000000\"}"
+// 2. "option:VOTE_OPTION_NO weight:\"1.000000000000000000\""
```

Comment:
VoteOption inside event is represented in these two type string, v1 and v1beta1 currently.

Reply:
I am still trying to understand cosmos SDK.

We are not able to use provided methods like:

```go
func (m *Vote) GetOptions() []*WeightedVoteOption {
	if m != nil {
		return m.Options
	}
	return nil
}
```